src/io/fstream.cpp

```diff
@@ -23,13 +23,13 @@ namespace fc {
    ofstream::ofstream()
    :my( new impl() ){}
 
-   ofstream::ofstream( const fc::path& file, int m )
+   ofstream::ofstream( const fc::path& file, std::ios_base::openmode m )
    :my( new impl() ) { this->open( file, m ); }
    ofstream::~ofstream(){}
 
-   void ofstream::open( const fc::path& file, int m ) {
+   void ofstream::open( const fc::path& file, std::ios_base::openmode m ) {
      const boost::filesystem::path& bfp = file; 
-     my->ofs.open( bfp, std::ios::binary );
+     my->ofs.open( bfp, m );
```

Comment:
@jmjatlanta look please fstream.hpp the default values are already there

https://github.com/bitshares/bitshares-fc/compare/0a90eff69778c0ef1d26f75d11d459ca172eff0c...71b6f154a62491c91a9dcdc29e9333e32f97f8d0#diff-aae0d97a407fcae9906054c8e1b9aa86R16

https://github.com/bitshares/bitshares-fc/commit/71b6f154a62491c91a9dcdc29e9333e32f97f8d0#diff-aae0d97a407fcae9906054c8e1b9aa86R19

isn't it correct ?

Reply:
I believe line 32 should read something like ``my->ofs.open( bfp, std::ios_base::out | std::ios_base::binary | m);``

That way, any existing code works just the way it used to.

Reply:
I made the same thing as you suggest and after that just checked my current solution and it's work the same, isn't it the same ?

Reply:
I will make as you suggest right now :-) just wanted to make as small changes as possible and thought that we have already default values

Reply:
done !

Reply:
It is almost the same, but not quite. Here is a contrived example: Think about the scenario of if someone passed only ``ios::ios_base::app``. Without the change above, the default values get replaced with only ``ios::ios_base::app``. The default values we provide would have been replaced.

By fixing the above, what the user sends this method gets appended to what we have, instead of replacing what we have.